### ISSUE TITLE
[Bug]:Failed to load collection

### ISSUE THREAD

**Issue by dzqoo** [milvus-io/milvus#18968]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.1.0
- Deployment mode(standalone or cluster):cluster
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus==2.1.0
- OS(Ubuntu or CentOS): centos
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

When I load collection, I got the following error:
![59103e92bb16645f6fef37dae788ffb](https://user-images.githubusercontent.com/8862305/187851476-f50b7437-28ff-4a4c-ada9-6d997439068c.jpg)
Here is all the logs:
[milvus-log.tar.gz](https://github.com/milvus-io/milvus/files/9467581/milvus-log.tar.gz)

**Comment by letian-jiang** [comment]
Hi @dzqoo, Could you please upload any parquet file (e.g. a non-sensitive column) for debugging? 

We suspect there is something wrong with the parquet header.

**Comment by letian-jiang** [comment]
Hi @dzqoo, have you ever modified the `insertBufSize` in `milvus.yaml`, which is 16MB by default?

Parquet has a default maximum row-group length of 67108864. Under the default setting, each parquet file may contain exactly one row-group. If the default size changes, this assumption may no longer hold.

**Comment by dzqoo** [comment]
> 

Thanks for applying.I have not changed the config.

**Comment by congqixia** [comment]
This problem is caused by marisa string index. By default, all binlog/index files are smaller than the max row group size(16MB by default). So the payload reader shall only read first element in parquet files. 
If the file is large enough (64MB shall be the config now), payload writer will split data into multiple row groups.
#19283 adapts the reading multiple raw group logic for fix this problem.
Maybe we shall split marisa index file? 
WDYGT @xige-16 @czs007  @xiaofan-luan 
Also, the file size limitation shall differs among different FS. Say HDFS shall prefer larger files?

**Comment by congqixia** [comment]
@dzqoo this problem shall be fixed in latest master and 2.1.4, could you please verify whether fix patches solve your issue?
/unassign
/unassign @letian-jiang 
/assign @dzqoo
